coral/src/app/features/dashboard/hooks/useDashboardData.ts

```diff
@@ -0,0 +1,88 @@
+import { useQuery } from "@tanstack/react-query";
+import {
+  DashboardsAnalyticsData,
+  getActivityLogForTeamOverview,
+  getDashboardStats,
+} from "src/domain/analytics";
+import { AuthUser, getAuth } from "src/domain/auth-user";
+import { getRequestsWaitingForApproval } from "src/domain/requests";
+
+type DashboardData = {
+  chartsData?: DashboardsAnalyticsData;
+  statsData: {
+    producerCount?: number;
+    consumerCount?: number;
+    totalTeamTopics?: number;
+    totalOrgTopics?: number;
+    pendingTopicRequests?: number;
+    pendingAclRequests?: number;
+    pendingSchemaRequests?: number;
+    pendingConnectorRequests?: number;
+  };
+};
+
+export const useDashboardData = (): {
+  data: DashboardData;
+  isLoading: boolean;
+  isError: boolean;
+  error: string;
+} => {
+  const {
+    data: chartsData,
+    isLoading: isLoadingChartsData,
+    isError: isErrorChartsData,
+  } = useQuery<DashboardsAnalyticsData, Error>(
+    ["getActivityLogForTeamOverview"],
+    {
+      queryFn: () => getActivityLogForTeamOverview(),
+    }
+  );
+
+  const {
+    data: statsData,
+    isLoading: isLoadingStatsData,
+    isError: isErrorStatsData,
+  } = useQuery(["getDashboardStats"], {
+    queryFn: getDashboardStats,
+  });
+
+  const {
+    data: pendingRequestsData,
+    isLoading: isLoadingPendingRequestsData,
+    isError: isErrorPendingRequestsData,
+  } = useQuery(["getRequestsWaitingForApproval"], {
+    queryFn: getRequestsWaitingForApproval,
+  });
+
+  const {
+    data: authUserData,
+    isLoading: isLoadingAuthUserData,
+    isError: isErrorAuthUserData,
+  } = useQuery<AuthUser | undefined>(["user-getAuth-data"], getAuth);
+
+  return {
+    data: {
+      chartsData,
+      statsData: {
+        ...statsData,
+        totalTeamTopics: authUserData?.totalTeamTopics,
+        totalOrgTopics: authUserData?.totalOrgTopics,
+        pendingTopicRequests: pendingRequestsData?.TOPIC,
+        pendingAclRequests: pendingRequestsData?.ACL,
+        pendingSchemaRequests: pendingRequestsData?.SCHEMA,
+        pendingConnectorRequests: pendingRequestsData?.CONNECTOR,
+      },
+    },
+    isLoading:
+      isLoadingChartsData ||
+      isLoadingStatsData ||
+      isLoadingAuthUserData ||
+      isLoadingPendingRequestsData,
+    isError:
+      isErrorChartsData ||
+      isErrorStatsData ||
+      isErrorAuthUserData ||
+      isErrorPendingRequestsData,
+    error: "Dashboard data could not be loaded",
```

Comment:
Is "Dashboard data could not be loaded" a good fallback? If someone gets this error not on the dashboard, it may be confusing. 

Also I'm thinking if we need to be more detailed with the error message anyway, depending on how the errors from the backend look like. 

The error notification would can pop up quite often on different pages, so it's harder for user to get the context of the error. If the message is vague, it may be hard to understand what is happening. 

I'm thinking about something like

```

`Error loading data for ending request and statistics: {errorFromBackend}`

```

(well, maybe a bit nicer)

Reply:
> Is "Dashboard data could not be loaded" a good fallback? If someone gets this error not on the dashboard, it may be confusing.

I mean, it is meant to be excusively used for the Dashboard, as the name `useDashboardData` implies, as well as its place in the project's hierarchy ^^'

And I thought that it would be a very unusual error, and would only pop up on the Dashboard page. It would then result in the page showing partial data (not having the elements which failed), and then people can reload, which would usually fix the problem.

I don't believe the API returns very useful errors for GET endpoints, as well.